ambry-api/src/main/java/com/github/ambry/quota/StorageQuotaService.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2020 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.quota;
+
+import com.github.ambry.account.Account;
+import com.github.ambry.account.Container;
+
+
+/**
```

Comment:
Its not clear to me whats the difference between `StorageQuotaService` and `StorageQuotaEnforcer`. This interface seems to have same methods as `StorageQuotaEnforcer`.

Reply:
StorageQuotaService is more like a wrapper, it's responsibility is to 1) choose different implementation of `StorageQuotaSource` `StorageUsageRefresher` and `StorageQuotaEnforcer` 2) glue them together and make it work.

But mainly the reason why I have `StorageQuotaService` and `StorageQuotaServiceFactory` is to use different implementations of those components. In closed source ambry, I would create a QuotaInStorageQuotaSource, which will get the quota from QuotaIn service, a HelixBasedDataPlanStorageUsageRefresher, which will get the monthly storage usage instead of total usage from helix. And probably a different storage quota enforcer.